samtranslator/model/api/api_generator.py

```diff
@@ -581,22 +581,36 @@ def _construct_record_sets_for_domain(
         self, domain: Dict[str, Any], api_domain_name: str, route53: Any
     ) -> List[Dict[str, Any]]:
         recordset_list = []
-        recordset = {}
 
+        recordset = self._construct_route53_routing_policy_properties(route53, {})
         recordset["Name"] = domain.get("DomainName")
         recordset["Type"] = "A"
         recordset["AliasTarget"] = self._construct_alias_target(domain, api_domain_name, route53)
         recordset_list.extend([recordset])
 
-        recordset_ipv6 = {}
         if route53.get("IpV6") is not None and route53.get("IpV6") is True:
+            recordset_ipv6 = self._construct_route53_routing_policy_properties(route53, {})
             recordset_ipv6["Name"] = domain.get("DomainName")
             recordset_ipv6["Type"] = "AAAA"
             recordset_ipv6["AliasTarget"] = self._construct_alias_target(domain, api_domain_name, route53)
             recordset_list.extend([recordset_ipv6])
 
         return recordset_list
 
+    def _construct_route53_routing_policy_properties(
+        self, route53_config: Dict[str, Any], recordset: Dict[str, Any]
+    ) -> Dict[str, Any]:
```

Comment:
I'd rather either make this a pure function (input/output, no mutations, perhaps `@staticmethod` to convey intention clearly), or get rid of the `return` so that it's more obvious it mutates inputs by design. Functions with side effects a little harder to reason about (and anecdotally more error-prone), so we try to minimize them, or when it makes sense make it abundantly clear.